src/actions/deviceActions.ts

```diff
@@ -264,30 +265,39 @@ export const open =
             DataManager().addData(cappedValue, b16 | prevBits);
             prevBits = 0;
 
-            if (getRecordingMode(getState()) === 'Scope') {
+            if (getRecordingMode(state) === 'Scope') {
+                const isRaisingEdge = state.app.trigger.edge === 'Raising Edge';
+                const isLoweringEdge =
+                    state.app.trigger.edge === 'Lowering Edge';
+
                 const validTriggerValue =
                     prevCappedValue != null &&
-                    prevCappedValue < getState().app.trigger.level &&
-                    cappedValue >= getState().app.trigger.level;
+                    ((isRaisingEdge &&
+                        prevCappedValue < state.app.trigger.level &&
+                        cappedValue >= state.app.trigger.level) ||
+                        (isLoweringEdge &&
+                            prevCappedValue > state.app.trigger.level &&
+                            cappedValue <= state.app.trigger.level));
+
```

Comment:
Consider it to make it a bit more readable 
```suggestion
if (isRaisingEdge){
 validTriggerValue = prevCappedValue != null && prevCappedValue < state.app.trigger.level &&
                         cappedValue >= state.app.trigger.level
}else if (isLoweringEdge){
   validTriggerValue = prevCappedValue != null && prevCappedValue > state.app.trigger.level &&
                             cappedValue <= state.app.trigger.level)
}
                    ((isRaisingEdge &&
                        prevCappedValue < state.app.trigger.level &&
                        cappedValue >= state.app.trigger.level) ||
                        (isLoweringEdge &&
                            prevCappedValue > state.app.trigger.level &&
                            cappedValue <= state.app.trigger.level));

```